### ISSUE TITLE
Permissions changed with version 0.6.1

### ISSUE THREAD

**Issue by NickNeck** [elixir-lang/elixir_make#44]
Hello,

with version 0.6.1, the permission of some files has changed.
```shell
$> tail -n 1 deps/elixir_make/hex_metadata.config
{<<"version">>,<<"0.6.0">>}.
$> ll -a deps/elixir_make
total 40
drwxr-xr-x   9 kruse  staff   288B Dec  2 17:07 .
drwxr-xr-x  79 kruse  staff   2.5K Dec  2 17:07 ..
-rw-r--r--   1 kruse  staff     0B Dec  2 17:07 .fetch
-rw-r--r--   1 kruse  staff    67B Dec  2 17:07 .formatter.exs
-rw-r--r--   1 kruse  staff   275B Dec  2 17:07 .hex
-rw-r--r--   1 kruse  staff   1.4K Dec  2 17:07 README.md
-rw-r--r--   1 kruse  staff   504B Dec  2 17:07 hex_metadata.config
drwxr-xr-x   3 kruse  staff    96B Dec  2 17:07 lib
-rw-r--r--   1 kruse  staff   596B Dec  2 17:07 mix.exs
...
$> tail -n 1 deps/elixir_make/hex_metadata.config
{<<"version">>,<<"0.6.1">>}.
$> ll -a deps/elixir_make
total 48
drwxr-xr-x  10 kruse  staff   320B Dec  2 17:09 .
drwxr-xr-x  79 kruse  staff   2.5K Dec  2 17:09 ..
-rw-r--r--   1 kruse  staff     0B Dec  2 17:09 .fetch
-rw-------   1 kruse  staff    67B Dec  2 17:09 .formatter.exs
-rw-r--r--   1 kruse  staff   275B Dec  2 17:09 .hex
-rw-r--r--   1 kruse  staff   153B Dec  2 17:09 CHANGELOG.md
-rw-------   1 kruse  staff   1.4K Dec  2 17:09 README.md
-rw-r--r--   1 kruse  staff   523B Dec  2 17:09 hex_metadata.config
drwxr-xr-x   3 kruse  staff    96B Dec  2 17:09 lib
-rw-------   1 kruse  staff   845B Dec  2 17:09 mix.exs
```
That caused some problems for me when I was trying to create a docker image from my app.

**Comment by whatyouhide** [comment]
@ericmj could this be caused by Hex in any way? I’m not sure about `mix.exs`, but `.fetch` is Hex's stuff 🧐

**Comment by ericmj** [comment]
`.fetch` is created by mix but the permissions are unchanged for that file.

The permissions are different because they were different on the publisher's machine when 0.6.1 was published compared to when 0.6.0 was published. This is because some packages require hex to preserve the permissions of the source files.

```
> tar -tvf elixir_make-0.6.0/contents.tar.gz
-rw-r--r--  0 0      0       10495 Jan  1  2000 lib/mix/tasks/compile.make.ex
-rw-r--r--  0 0      0          67 Jan  1  2000 .formatter.exs
-rw-r--r--  0 0      0         596 Jan  1  2000 mix.exs
-rw-r--r--  0 0      0        1474 Jan  1  2000 README.md
> tar -tvf elixir_make-0.6.1/contents.tar.gz
-rw-r--r--  0 0      0       11656 Jan  1  2000 lib/mix/tasks/compile.make.ex
-rw-------  0 0      0          67 Jan  1  2000 .formatter.exs
-rw-------  0 0      0         845 Jan  1  2000 mix.exs
-rw-------  0 0      0        1474 Jan  1  2000 README.md
-rw-r--r--  0 0      0         153 Jan  1  2000 CHANGELOG.md
```

If these permissions cause issues then 0.6.2 should be republished with fixed permissions and 0.6.1 be retired. The permissions seems to be correct in the git repository so it seems the permissions only were wrong on the publishers machine when 0.6.1 was published for some reason.

```
> git clone https://github.com/elixir-lang/elixir_make
Cloning into 'elixir_make'...
remote: Enumerating objects: 34, done.
remote: Counting objects: 100% (34/34), done.
remote: Compressing objects: 100% (17/17), done.
remote: Total 346 (delta 7), reused 21 (delta 4), pack-reused 312
Receiving objects: 100% (346/346), 54.24 KiB | 730.00 KiB/s, done.
Resolving deltas: 100% (113/113), done.
> ls -Rl elixir_make
total 32
-rw-r--r--  1 ericmj  staff   153 Dec  2 22:45 CHANGELOG.md
-rw-r--r--  1 ericmj  staff  1474 Dec  2 22:45 README.md
drwxr-xr-x  3 ericmj  staff    96 Dec  2 22:45 lib
-rw-r--r--  1 ericmj  staff   845 Dec  2 22:45 mix.exs
-rw-r--r--  1 ericmj  staff  1315 Dec  2 22:45 mix.lock
drwxr-xr-x  5 ericmj  staff   160 Dec  2 22:45 test

elixir_make/lib:
total 0
drwxr-xr-x  3 ericmj  staff  96 Dec  2 22:45 mix

elixir_make/lib/mix:
total 0
drwxr-xr-x  3 ericmj  staff  96 Dec  2 22:45 tasks

elixir_make/lib/mix/tasks:
total 24
-rw-r--r--  1 ericmj  staff  11656 Dec  2 22:45 compile.make.ex

elixir_make/test:
total 8
drwxr-xr-x  3 ericmj  staff  96 Dec  2 22:45 fixtures
drwxr-xr-x  3 ericmj  staff  96 Dec  2 22:45 mix
-rw-r--r--  1 ericmj  staff  15 Dec  2 22:45 test_helper.exs

elixir_make/test/fixtures:
total 0
drwxr-xr-x  3 ericmj  staff  96 Dec  2 22:45 my_app

elixir_make/test/fixtures/my_app:
total 8
-rw-r--r--  1 ericmj  staff  137 Dec  2 22:45 mix.exs

elixir_make/test/mix:
total 0
drwxr-xr-x  3 ericmj  staff  96 Dec  2 22:45 tasks

elixir_make/test/mix/tasks:
total 16
-rw-r--r--  1 ericmj  staff  7892 Dec  2 22:45 compile.make_test.exs
```

**Comment by whatyouhide** [comment]
Ah, good to know @ericmj. I think I have published the broken version but I wouldn't even know where to start figuring out how the hell permissions changed there 🤔 Anyways, I released 0.6.2 and retired 0.6.1. Thanks for the help! 💟
